### PR TITLE
[JITM] Support in other screens infrastructure

### PR DESCRIPTION
**Pull request by toupper**
### Description

We want to refactor here the JITM implementation to add new placements in screens other than the dashboard. For that, we move the JITM retrieval logic out of `DashboardViewModel` into its own class, so it can be reused for new placements. Furthermore, we also add an `enum` to gather the placement screens, hiding the map between them and the message path string inside the new `JustInTimeMessagesProvider`. 

Lastly, we convert the logic to `async/await`, making it easier to load the new messages and synchronize the products onboarding priority logic. In order to ensure that UI code is running on the main thread, we use  `Task` with @ `MainActor`.

### Testing instructions

Ensure that the JITM in the Dashboard is still working as expected. Be sure to cover all cases: showing the JITM, dismissal, refresh, and orientation changes. Furthermore, unit tests should pass.

### Screenshots

---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/JustInTimeMessages/JustInTimeMessagesProvider.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    init(stores: StoresManager = ServiceLocator.stores,
+         analytics: Analytics = ServiceLocator.analytics) {
+        self.stores = stores
+        self.analytics = analytics
+    }
+
+    func loadMessage(for screen: JustInTimeMessagesSourceScreen, siteID: Int64) async throws -> JustInTimeMessageAnnouncementCardViewModel? {
+        guard let source = appScreenJitmSourceMapping[screen] else {
+            return nil
```
Comment (joshheald) [inline WooCommerce/Classes/JustInTimeMessages/JustInTimeMessagesProvider.swift:24]: Should we log something here? Maybe not an error, just info?

**Review by joshheald — APPROVED** [review]
Looks good to me, but it would be great for @rachelmcr to take a look over as well for the products onboarding stuff. I don't have a test store set up for that. @rachelmcr if you don't have time, let me know and I'll make a store, or if you could add me to one of yours?

One question about logging.

More broadly... the `Task { @MainActor` needing to be added all over the place... is there a neater way? E.g. is it possible/desirable to specify `@MainActor` on the `syncAnnouncements` function in the view model, or somehow on the publisher, so that we're on the main thread in the DashboardViewController without having to constantly check?

**Review by rachelmcr — APPROVED** [review]
> Looks good to me, but it would be great for @rachelmcr to take a look over as well for the products onboarding stuff.

Sure thing! I did a quick test on a store with no products (thus eligible for products onboarding) and it worked fine there. 👍 I took just a quick look over the code, as well, and nothing jumped out at me as a concern.

Let me know if you'd like me to add you to my test store for ease of future testing! It also works to just spin up a temporary new store (e.g. with Jurassic Ninja) since new stores are eligible for this onboarding.

File: WooCommerce/Classes/JustInTimeMessages/JustInTimeMessagesProvider.swift (hunk 1 above)
Comment (toupper) [reply WooCommerce/Classes/JustInTimeMessages/JustInTimeMessagesProvider.swift:24]: Sure, added in fa90dafc440c4f1699a63e3267a8abcad5820320

**Comment by toupper** [comment]
Thanks for the review and your comment @joshheald. I removed the `@MainActor` when calling `syncAnnouncements` because it was not necessary, we are actually calling the UI code on the main thread. I prefer to leave the view model without @MainActor so it can run the non-UI code on background threads. That way, the view controller is in charge of specifying the need of running the code on the main thread.

**Pull request merged**